lib/Doctrine/Annotations/Metadata/Constraint/InvalidValue.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Metadata\Constraint;
+
+use function gettype;
+use function implode;
+use function sprintf;
+
+final class InvalidValue extends ConstraintNotFulfilled
+{
+    /**
+     * @param mixed[] $allowedValues
+     * @param mixed   $value
+     */
+    public static function new(array $allowedValues, $value) : self
+    {
+        // TODO: Describe values
+        return new self(
+            sprintf(
+                'Invalid value "%s" for allowed values: "%s".',
+                gettype($value),
+                implode(',', $allowedValues)
```

Comment:
`$allowedValues` is `mixed`, but `implode()` assumes strings.

Reply:
Looks like my TODO is about it. 
Should we describe values or remove them from message?
Do you know a lib that does this?